### DISCUSSION TITLE
lsmod error

### DISCUSSION THREAD

**Discussion by xiaofeng0303** [cloud-hypervisor/cloud-hypervisor#2700] [Q&A]
I start a vm(centos7)by cloud-hypervisor,
when exec `lsmod` in vm output an error:
```
 libkmod: kmod_module_new_from_loaded: could not open /proc/modules: No such file or directory
```
how do i know this guest kernel modules

**Comment by iggy** [comment]
You didn't say what kernel you were using or how you started cloud-hypervisor.

**Reply by xiaofeng0303** [reply]
kernel is : vmlinux-virtiofs.container (kata  build kernel)
disk: centos7.4
cmd:
./cloud-hypervisor --kernel ./vmlinux-virtiofs.container --console off --serial tty --disk path=/opt/image-centos-74.raw --cmdline "root=/dev/vda1 panic=1 no_timer_check noreplace-smp rootflags=data=ordered,errors=remount-ro ro rootfstype=ext4" --cpus boot=4 --memory size=1024M --vsock cid=3,socket=/tmp/test.vsock,id=test1 --rng

**Comment by arafatms** [comment]
Maybe you did not set some several configuration when building a custom kernel，such as CONFIG_MODULES

**Reply by scuzhanglei** [reply]
when I boot from a centos cloud image ,always get error `Failed to find EFI partition: HeaderNotFound`,  as centos image is MBR partition with out EFI partition, how could you boot for a centos image?
